static/app/utils/number/ladder.tsx

```diff
@@ -0,0 +1,63 @@
+import orderBy from 'lodash/orderBy';
+
+type Rung<T> = [threshold: number, value: T];
+
+/**
+ * Class that represents a value ladder. Given a ladder of increasing thresholds,
+ * the ladder can match an incoming value against the known intervals.
+ *
+ * @example
+ * ```javascript
+ * // Selecting a quota based on team size
+ * const ladder = new Ladder([
+ *   [0, 10],
+ *   [10, 100_000],
+ *   [30, 1_000_000],
+ * ]);
+ *
+ * const quota = ladder.step(0); // Quota for small teams is 10
+ * const quota = ladder.step(12); // Quota for a 10-20 person team is 100,000
+ * const quota = ladder.step(120); // Quota for a 30 person or higher team is 1,000,000
+ * ```
+ */
+export class Ladder<T> {
+  rungs: Rung<T>[];
+
+  // At least one rung is required by the type
+  constructor(rungs: [Rung<T>, ...Rung<T>[]]) {
+    if (!rungs.some(rung => rung[0] === 0)) {
+      throw new Error('At least one rung in the ladder must start at 0');
+    }
```

Comment:
How about removing this requirement and just deferring to the user to determine the min? You get a more flexible API that way and can simply defer to the user to determine the ladder ranges

Reply:
Agreed 👍🏻